### PR TITLE
Specify org-wide whitelists for prow plugins

### PR DESCRIPTION
**Pull request by foxish**
The code is not DRY and we should find a way to clean it up at some point (now?).

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/test-infra/1079)

### CODE REVIEW

**Comment by spxtr** [comment]
We can use this function to make it nicer:

```
// pluginsForRepo returns a list of plugins that are enabled on a given repository.
func (pa *PluginAgent) pluginsForRepo(org, repo string) []string
```

and then in `StatusEventHandlers` and the like change `for _, p := range pa.ps[repo] {` to `for _, p := range pa.pluginsForRepo(org, repo) {`.

**Comment by foxish** [comment]
@spxtr Made that change, and added a test. PTAL.

File: prow/plugins/plugins.go [hunk 1]
```diff
@@ -115,43 +115,58 @@ func (pa *PluginAgent) Start(path string) error {
...
 			hs[p] = h
 		}
 	}
+
 	return hs
 }
+
+// getPlugins returns a list of plugins that are enabled on a given (org, repository).
+func (pa *PluginAgent) getPlugins(owner, repo string) []string {
+	// TODO: use a set here to avoid repetitions?
```
Comment (spxtr) [inline prow/plugins/plugins.go]: Instead, we can add a unit test that validates `../plugins.yaml`, similar to the test for `jobs.yaml`. Can be done later.

**Comment by foxish** [comment]
In a different pass, I imagine we want to have some other bot commands (/lgtm?) also be org-wide.

**Comment by spxtr** [comment]
Looks good to me. We'll get some errors showing up if they don't have the cla label, but we can deal with that in a future pass.

Feel free to merge and update after fixing the trivial conflict.

**Comment by foxish** [comment]
@spxtr Also added kubernetes incubator to the list of repos where CLA is to be enabled. We need to turn setup the webhook on kubernetes-incubator for it to work.

**Pull request merged**